CryptoPrices/Sources/Data/Sources/DataTestHelpers/Mock/MockCoinAPI.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  MockCoinAPI.swift
+//  Data
+//
+//  Created by Doan Thieu on 09/12/2022.
+//
+
+import NetworkCore
+
+// TODO: - Set up code generator for generating mock types (e.g. Sourcery)
+
+public class MockCoinAPI: CoinAPIProtocol {
```

Comment:
Why don't we apply `Sourcery` right away? This code is going to be removed once we have `Sourcery` in place, which is a bit of wasted effort here to imo 😅

Reply:
The reason is that this PR is quite large, so I'd like to focus on how we would implement the first backend task (with the very first unit tests to be written as well) 🙏 

Speaking of a code generator for stub and mock, `Sourcery` is great and I might use it. But I'm still looking at other alternatives like [Mockingbird](https://github.com/birdrides/mockingbird), [Cuckoo](https://github.com/Brightify/Cuckoo), ... The idea is that not only we want to use the popular, widely-used and familiar tools but also we want to broaden the options, and solutions so we ourselves can have more skills under our belt 🙏 
 
Also, the number of mocks I have to write by hand is quite small (`MockAPI` and `MockCoinRepository`), so I just went ahead and added them first with considering replacing them later (hence the `TODO` marks).

Reply:
Noted, thank you for your though process there, I really like the idea that this project can be a good playground for us to experiment new tools 👍 Let's keep it as-is and hold the Sourcery integration for now 💪, and let's have a discussion to decide which mocking tool we would like to use for the project later then 🚀